### PR TITLE
Refactor examples folder, removed Vector3.hpp

### PR DESCRIPTION
**Pull request by wesleycsj**

### CODE REVIEW

**Review by ryukinix — CHANGES_REQUESTED** [review]
Since you preferred create sub-directories, you need now to fix the Makefile too. The variable EXAMPLES should be fixed with a similar definition of SRCS using find. The expected behavior is: running `make examples` should run all examples. Or make check should compile all examples etc

Since now we have class of examples too, which is far good, we need use properly too. `collision/square-moving.cpp` originated from `lerax-square.cpp` do not deal with collision. We need to put it in another place.

That example seems more related to... Maybe input? Basics? The following characteristics are covered:

- GameLoop
- Rect Rendering 
- Input Handling
- Vector2 usage with cos angle etc

It seems really related to "basics" usage of the API.

The rest it seems good to me.

**Comment by wesleycsj** [comment]
Since i remember, didn't you made the entire frame as a rect and tested if the square was colliding with that?

**Comment by ryukinix** [comment]
> Since i remember, didn't you made the entire frame as a rect and tested if the square was intersecting with that?

Sorry, we are not talking about `lerax-square-platform.cpp` as I said, which indeed does that and it's corrected related to collision. The problem is `lerax-square.cpp` (square moving)

Reading the `lerax-square.cpp`, as you can see in that image: 

image

Didn't even include the collider system.

**Comment by wesleycsj** [comment]
No, i was asking exactly for `lerax-square.cpp`. Anyway, i'll keep it in the `basics` folder. Sure we can make good usage of this naming in the future for engine basic concepts.

**Comment by ryukinix** [comment]
Thanks!

File: Makefile [hunk 1]
```diff
@@ -7,7 +7,7 @@ MAIN        = $(SRC_DIR)/main.cpp
 SRCS        = $(shell find $(SRC_DIR) -name "*.cpp" -type f)
 LIB_SRCS    = $(filter-out $(MAIN), $(SRCS))
 TESTS       = $(wildcard tests/*.cpp)
-EXAMPLES    = $(wildcard examples/*.cpp)
+EXAMPLES    = $(shell find examples -name "*.cpp" -type f)
```
Comment (ryukinix) [inline Makefile:10]: That fixes the problem I mentioned earlier, @wesleycsj

Comment (rafaelcn) [reply Makefile:10]: Beautiful.

Comment (ryukinix) [reply Makefile:10]: Oh, you are alive @rafaelcn 

@wesleycsj was asking for you another day.

**Review by ryukinix — APPROVED** [review]
Modifications requested made. Approved.

**Pull request merged**